src/main/groovy/com/electriccloud/pluginwizardhelp/DataSlurper.groovy

```diff
@@ -228,6 +211,62 @@ def procedure(name, closure) {
         procedureMetadata
     }
 
+    Map readProcedureMetadataFromDsl(String procedureDsl, String procedureFolderName = "No procedure name passed"){
+
+        def dslCode = """
+  this.pluginDir = '${this.pluginFolder}'
+  this.collectedData = [:]
+  
+  def collect(String type, Map object){
+    if (this.collectedData[type] == null){
+      this.collectedData[type] = []
+    }
+    this.collectedData[type] << object
+  }
+  
+  def procedure(params, name, closure) {
+      def retval = [:]
+      retval.name = name
+      retval.description = params.description
+      
+      // *unexpected* : Collecting closure data in other methods 
+      closure.call()
+      collect('procedure', retval)
+  }
+  
+  def procedure(name, closure) {
+      def retval = [:]
+      retval.name = name
+      retval.description = ''
+      
+      // *unexpected* : Collecting closure data in other methods 
+      closure.call()
+      collect('procedure', retval)
+  }
+  
+  void formalOutputParameter(params, name) {
+      def retval = [:]
+      retval.name = name
+      retval.description = params.description
+      collect('outputParameters', retval)
+  }
+  
+  void step(Object... args) {}
+  void property(Object... args) {}
```

Comment:
There is a possibility of getting actualParameter or formalParameter call

Reply:
Changed to methodMissing()